activemq-stomp/src/test/java/org/apache/activemq/transport/stomp/StompTestSupport.java

```diff
@@ -342,6 +342,8 @@ protected void addTranportConnectors() throws Exception {
             autoNioSslPort = connector.getConnectUri().getPort();
             LOG.debug("Using auto+nio+ssl port " + autoNioSslPort);
         }
+
+        connector.setAllowLinkStealing(true);
```

Comment:
That one can help in the case of a race condition because it can reuse the link from another client.

Reply:
Note: This closes the _other_ client and allows the new connection to use the clientId. 

While this would work, these smells more like a hack vs a fix. Why/Where is the other client connection coming from?

Reply:
I totally agree that this is a workaround rather than a proper fix. I'm just exploring different options.

> Why/Where is the other client connection coming from?

This is the right question! I'm puzzled by this and I'm not clear how to understand what test/client keeps the connection open as it is not reproducible in my local.

Reply:
Link stealing did not help:
https://ci-builds.apache.org/job/ActiveMQ/job/ActiveMQ/job/PR-1292/3/cloudbees-pipeline-explorer/#line-12035